clusterous/cluster.py

```diff
@@ -1177,6 +1276,9 @@ def rm_nodes(self, num_nodes, node_name):
         self._logger.info('Removing {0} nodes of type "{1}"...'.format(actual_num_to_remove, node_name))
 
         success = self._terminate_instances_and_wait(conn, ids_to_remove)
+        if spot_requests_to_cancel:
+            for request_id in spot_requests_to_cancel:
+                conn.cancel_spot_instance_requests(request_id, dry_run=False)
```

Comment:
In terminate_cluster, you use the same cancel_spot_instance_requests() method to destroy the entire spot request, not only certain spot instances. I haven't yet tested, but are you sure this is correct?